### DISCUSSION TITLE
Can Syncplay piggy back on Matrix protocol ?

### DISCUSSION THREAD

**Discussion by rishacha** [Syncplay/syncplay#648] [Q&A] [answered]
I wonder if there's a way to add a plugin to communicate syncplay sync updates over matrix protocol

**Comment by Et0h** [comment]
If people want to use other software at the same time as Syncplay for communication and/or planning then they are free to do so. I personally use Mumble for VoIP alongside Syncplay. However, we want to keep Syncplay as simple as possible and so there are no plans for Syncplay to support communication integration with messaging/VoIP/communications protocols and systems such as Matrix, IRC, Discord or Mumble.

**Reply by rishacha** [reply]
I think I might've framed the question incorrectly. The aim was to communicate the timestamps, play/pause signals and messages over matrix protocol instead of using syncplay's protocol.

**Reply by Et0h** [reply]
Why would be the benefits of this over using the existing Syncplay protocol?

**Reply by rishacha** [reply]
So currently, I am not using a publicly hosted syncplay server (probably because I couldn't get them to work). I setup my server on my local, use NordVPN's meshnet and then host a local syncplay server in this private network to get the job done. If instead, I could send the same signals over using some XMPP chat server that's publicly hosted (or private for that matter), I don't have to bother hosting a server (or setting up a private network). It would make life easier

**Reply by Et0h** [reply, marked as answer]
Thanks for the more detailed explanation - that helps me better understand your proposal .

I had a quick look for a review of the Matrix protocol and came across https://anarc.at/blog/2022-06-17-matrix-notes/

Based on what that blog post says it seems to me that it would not be advisable to tunnel the Syncplay protocol over Matrix due to issues including rate limiting, latency and privacy (let alone the complexity).

It might be possible to design a sync solution around the Matrix protocol, but that would need to be a programme specifically designed around that protocol which Syncplay is not.

As such, it looks like you will need to continue to run your own Syncplay server for the foreseeable future.

**Reply by Et0h** [reply]
On a separate note, it might be would be great if you could write a step-by-step guide on how to use Syncplay with meshnet as it could help other users who are in your situation.

**Reply by rishacha** [reply]
> Thanks for the more detailed explanation - that helps me better understand your proposal .
> 
> 
> 
> I had a quick look for a review of the Matrix protocol and came across https://anarc.at/blog/2022-06-17-matrix-notes/
> 
> 
> 
> Based on what that blog post says it seems to me that it would not be advisable to tunnel the Syncplay protocol over Matrix due to issues including rate limiting, latency and privacy (let alone the complexity).
> 
> 
> 
> It might be possible to design a sync solution around the Matrix protocol, but that would need to be a programme specifically designed around that protocol which Syncplay is not.
> 
> 
> 
> As such, it looks like you will need to continue to run your own Syncplay server for the foreseeable future.

This is exactly the discussion I wanted to trigger. Thank you @Et0h  for patiently answering my queries.

**Reply by rishacha** [reply]
> On a separate note, it might be would be great if you could write a step-by-step guide on how to use Syncplay with meshnet as it could help other users who are in your situation.

Definitely! Where do you want me to publish this guide? This could be a boon for long-distance couples/friends/groups since it's lag-free and consumes way less bandwidth than Plex over Meshnet. And Meshnet is free for the foreseeable future afaik.

**Reply by Et0h** [reply]
> > On a separate note, it might be would be great if you could write a step-by-step guide on how to use Syncplay with meshnet as it could help other users who are in your situation.
> 
> Definitely! Where do you want me to publish this guide? This could be a boon for long-distance couples/friends/groups since it's lag-free and consumes way less bandwidth than Plex over Meshnet. And Meshnet is free for the foreseeable future afaik. 

I think the wiki makes sense as we already have 3 guides on there which are  https://github.com/Syncplay/syncplay/wiki/TLS-support and https://github.com/Syncplay/syncplay/wiki/Guide-to-adding-support-for-a-new-media-player-to-Syncplay and https://github.com/Syncplay/syncplay/wiki/Guide-to-translating-Syncplay-into-another-language

I can look into giving you wiki editing access or you can just write the guide here and I'll copy it over.

**Reply by rishacha** [reply]
# Guide on how to use SyncPlay server on Nord VPN's MeshNet

## Background 
Before we begin, if you need a background on MeshNet, please refer to this [page](https://nordvpn.com/meshnet/). For simplicity, I will just refer to MeshNet as a LAN connection between devices enabled using NordVPN's client. 

As of today, [Meshnet is free](https://nordvpn.com/blog/why-is-meshnet-free)

## Setup Guide

1. All parties must connect and register their devices in your Meshnet
<img width="904" alt="image" src="https://github.com/Syncplay/syncplay/assets/30197357/42a42107-8578-459c-98bf-81b95f2d0837">

2. Start your Syncplay server using CLI
<img width="731" alt="image" src="https://github.com/Syncplay/syncplay/assets/30197357/f71f11dc-5e25-4107-a44d-765f4da1cbba">

3. Share your IP with everyone
<img width="841" alt="image" src="https://github.com/Syncplay/syncplay/assets/30197357/aac5cc2d-d18a-41df-8ed4-4f5bdf287e4a">

4. Open Syncplay Client and connect using the IP:Port
<img width="821" alt="image" src="https://github.com/Syncplay/syncplay/assets/30197357/b99dbf82-d6c3-4295-b918-7b5ee7eb993f">


5. Voila, you should be connected !


Enjoy your lag-free experience (much much better than Plex Media Server on Meshnet)

**Comment by Et0h** [comment]
@rishacha Excellent. The guide is now available at https://github.com/Syncplay/syncplay/wiki/Guide-on-how-to-use-SyncPlay-server-on-NordVPN's-MeshNet which I mentioned in the FAQ at https://syncplay.pl/guide/server/ along with the other guides on how to run the Syncplay server in different environments.
